### DISCUSSION TITLE
DSPL=time

### DISCUSSION THREAD

**Discussion by gregorpesek** [gopro/labs#168] [General]
Think it would be very useful for everyone if you included this¹ information to the [Personalize via QR Code](https://gopro.github.io/labs/control/owner/) because it's easy to miss when it is only mentioned once.

¹DSPL=time, this will control the amount of time messages are displayed. For users who want there own information displayed longer. The default is 1 second. DSPL=1 thru 9 is in seconds. DSPL = 10 thru 9999 is in milliseconds. So for much faster messages set DSPL to 100. Set this before setting the owner information, as metadata commands are processed in the order they are stored."

**Comment by gregorpesek** [comment]
Btw what am I doing wrong? I put 1. !RESET 2. DSPL=4 3. !MOWNR="information" and when I turn on the camera the info is only displayed for 1 second instead of 4.

Edit: think what happened is I put first 2 seconds "DSPL=2" and now it's stuck at 2 because "!RESET - erase all your permanent metadata (anything that used !Mxxxx command.)" and cannot be change unless factory reset?

**Reply by dnewman-gpsw** [reply]
The needed command is !MDSPL=4 (not just DSPL=4) and it works in my testing just now.  You can change it is as often as needed.  Take a shot video or photo, after you set this metadata, then extract the metadata using https://gopro.github.io/labs/control/metadata/  to see if you have everything setup correctly.
![image](https://user-images.githubusercontent.com/10856084/208499329-928ceb21-f5b0-41a8-8dda-54bf3e3e36af.png)

**Reply by gregorpesek** [reply]
I'm very shocked how this is not already stated on github under /control/extensions.. Why don't you just type to use the !MDSPL in the first place? I mean how do you think people will automatically assume that !M has to be added upfront? I mean it makes sense now that you said it but it makes absolutely no sense to include it when it says "DSPL=x will increase the display time for information.."

Please edit it here 
https://gopro.github.io/labs/control/extensions/
and add the same text to
https://gopro.github.io/labs/control/owner/

I was wondering why this command never worked since it was introduced and I just never bothered to ask about it.. I really do appreciate your contribution to labs, but I'm speechless on this one. I really do appreciate your contribution to labs.

BTW it does work for me now aswell thank you for helping me figure it out, and THANK YOU for everything you contribute to labs! 

Please fulfill my request so other people will have an easier time 🙏

**Reply by dnewman-gpsw** [reply]
I guess you skipped over the first section:
"All metadata QR commands are written in the form oM**wxzy**=value(s) or !M**wxzy**=value(s) – where the four character code (4CC) wxzy is under your control, along with the data it stores. The **!M** version will permanently store the metadata, and the **oM** will store it for only this power-on session. Metadata is available to flag your files for special uses, or just to label the [camera owner](https://gopro.github.io/labs/control/extensions/owner). Some particular 4CCs will also change camera behavior and/or enable features"

**Reply by gregorpesek** [reply]
1. No, I did not skip the part. 
If you asked me : "Do you know that to make the camera remember a command for next sessions you have to include another command?" I'd tell you "I know but I can't remember the exact command because I never use it for myself.. "
2. oM is not a must and I never used it because the commands will work just as fine for the session I'm on.. 
3.  Well the only time I use the command !M is when I use the AUTO GENERATOR to generate for me the owner's information and that's it! I never have to 
 type out !M manually for anything  else
4. Can we both agree that the DSPL command has ABSOLUTELY NOO use if it's not being remembered for the next session by the camera? So in my perspective DSPL command was an exception where !M is not required.. 


I mean seriously if you think your way, why is it when generating owner's information the generator includes !M by it self? Why not make it without and have the peasant like me figure it self out? 

DAN people like me who enjoy labs every once in a while don't remember almost every command as someone like you do. Labs is huge, there are just to many commands to be remembered. To be honest, I actually don't remember any of the commands from the top of my head because I always use the generators to do it for me or I have it saved somewhere.. And when on the website it states "X command will do THIS"  and I generate the code and it does not do the thing that it says it will do, I will get thrown off.

You have to understand that most people like me are peasants when it comes to using labs..

I'm not trying to argue. I'm just giving you my perspective as feedback.

Cheers!

**Reply by dnewman-gpsw** [reply]
The extensions page works like owner, just scroll to the bottom and use the provided controls.

**Reply by gregorpesek** [reply]
Not sure if you responded to the right thread, did you?

**Reply by dnewman-gpsw** [reply]
I did, correct thread. Everything is as planned, as the 4CCs need to be documented for reading as well as writing.  As this is all on GitHub, you are welcome to create your own versions of these pages.

**Reply by mattez** [reply]
@gregorpesek and for others: 

You correctly used `!RESET`.

Then you simply read these `!MDSPL=5` and `!MOWNR="My info text"` via [QR code](https://gopro.github.io/labs/control/extensions/#addcmd). 

or chained together like:

```
!MDSPL=5!MOWNR="My info text"
```
See ready QR code here:
[https://gopro.github.io/labs/control/set/?cmd=!MDSPL=5!MOWNR="My info text"](https://gopro.github.io/labs/control/set/?cmd=!MDSPL=5!MOWNR%3D%22My%20info%20text%22)

**Reply by mattez** [reply]
I found something new for me: see https://github.com/gopro/labs/discussions/85#discussioncomment-7571007

@gregorpesek wrote:

> Another trick is use additional owner info, using OWNS and OWNT etc.
...
> Try
> !MOWNS="message 1"
> !MOWNT="message 2"
> !MOWNU="message 3"

Interesting hack :) I am going to try that... (I wonder what's that gonna do?)
